search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/ProcessInstanceFilterTransformer.java

```diff
@@ -97,6 +98,10 @@ public ArrayList<SearchQuery> toSearchQueryFields(final ProcessInstanceFilter fi
       queries.add(term(PARTITION_ID, filter.partitionId()));
     }
 
+    if (filter.tags() != null && !filter.tags().isEmpty()) {
+      queries.add(and(filter.tags().stream().map(tag -> term(TAGS, tag)).toList()));
+    }
```

Comment:
💭 I have to admit, this is not intuitive to me 😅 I basically define "tags must be equal to `x` and tags must be equal to `y` and ...". Could we add a short comment here on what we want to achieve and why this works? A link to official docs also works.

Reply:
Just a code comment 👍